src/services/component.ts

```diff
@@ -21,31 +20,27 @@ type ComponentPropsGetter<TComponent extends Component> = () => MaybePromise<Com
  * @param component The component to mount
  * @param props A callback that returns the props or attributes to bind to the component
  * @returns A component
- *
- * @example
- * ```ts
- * import { createRoute, component } from '@kitbag/router'
- *
- * export const routes = createRoute({
- *   name: 'User',
- *   path: '/',
- *   component: component(User, () => ({ userId: 1 }))
- * })
- * ```
  */
 export function component<TComponent extends Component>(component: TComponent, props: ComponentPropsGetter<TComponent>): Component {
   return defineComponent({
     name: 'PropsWrapper',
     expose: [],
     setup() {
       const values = props()
+      const instance = getCurrentInstance()
 
       return () => {
         if (values instanceof Error) {
           return '' 
         }
 
         if (isPromise(values)) {
+          // there isn't a way to check if suspense is used in the component without accessing a private property
+          // @ts-expect-error
+          if(instance?.suspense) {
+            return h(suspenseAsyncPropsWrapper(component, values))
+          }
```

Comment:
@stackoverfloweth this is the best way I could figure out to check if the component is within a suspense boundary. This is a private part of the vue api. But if `suspense` is `null` that means there's no suspense boundary.